### PR TITLE
Add new attribute 'supported' in pyproject.toml

### PR DESCRIPTION
**Pull request by qlchan24**
### Changes:
Add additional metadata attribute `supported = false` in the `pyproject.toml` for ios and android if the GUI framework is PySide2 or PPB.

Formattting of the toml file generated with Briefcase `new` command with Pyside2 selected:
gui_pyside2

Formattting of the toml file generated with Briefcase `new` command with Toga selected:
gui_toga

## Problem solved:

Ensures that the `pyproject.toml` file generated from this template using the briefcase `new` command will contain a flag to disallow creation of mobile apps with unsupported GUI through the briefcase `create` or `update` command. More details provided in the related issue below.

## Related issue:
Briefcase issue #539: [Provide better warnings when an app *can't* be deployed to mobile platforms](https://github.com/beeware/briefcase/issues/539)

## PR Checklist:

- [x] All new features have been tested
- [x] All new features have been documented
- [x] I have read the **CONTRIBUTING.md** file
- [x] I will abide by the code of conduct

### CODE REVIEW

**Review by freakboy3742 — CHANGES_REQUESTED** [review]
This looks really good; The one suggestion I have is that for the cases where `supported=False`, there's no point including a requires block as well. As Toga is the only platform that realistically supports iOS and Android at present, I think it's fine to say `if toga: ... else: supported=False`

**Comment by qlchan24** [comment]
Thank you. I have changed the conditional tags as suggested and tested it out for the latest commit.

**Review by freakboy3742 — APPROVED** [review]
👍 Looks great! I'll merge this change at the same time I merge the related Briefcase PR.

**Comment by freakboy3742** [comment]
Correction - I'll need to wait until the next formal Briefcase release before merging. We're due for a new release - I'll aim to do this around the new year.

**Comment by qlchan24** [comment]
Sure, thanks for the update!

**Comment by freakboy3742** [comment]
My sincere apologies - Briefcase *did* receive a release in early 2021, but I forgot to I completely forgot to merge this PR. Better late than never? 😬

**Pull request merged**
